ui/src/common/components/__tests__/CitationSummaryGraph.test.jsx

```diff
@@ -340,7 +356,8 @@ describe('CitationSummaryGraph', () => {
         loadingCitationSummary={false}
         error={null}
         onSelectBarChange={onSelectBarChange}
-      />
+      />,
+      { disableLifecycleMethods: true }
```

Comment:
I get that you were having some problems, but this maybe not good in the future when we put more logic in the lifecycle methods. So maybe we could just mock `getBoundingClientRect` for all tests?... like you did in the container, but in `setupTests` so that we don't need to put it or disable lifecycle methods everywhere.

Reply:
The issue that I was having was not with mocking getBoundingClientRect, it was with mocking the ref (which doesn't work when using shallow), this is why I added the disable lifecycle methods. I agree that it shouldn't be done this way but mocking the getBoundingClientRect wouldn't solve this issue